modules/replication/dockerimage-replication/get_list_eks_images.py

```diff
@@ -151,32 +153,81 @@ def main() -> None:
                 if registry:
                     image = f"{registry}/{repository}"
                 if tag:
-                    image += f":{tag}"
+                    image += f":{tag}"  # type: ignore
 
                 logger.debug("\t\t%s", image)
-                images.append(image)
+                images_wip_list.append(str(image))
 
         logger.debug("\tCustom chart values:")
         logger.debug("\t\t%s", json.dumps(custom_chart_values[workload]))
+    return custom_chart_values
+
+
+def main() -> None:
+    """Main handler"""
+    args = parse_args(sys.argv[1:])
 
-    sorted_images = sorted(set(images))
+    logger.info("EKS version: %s", args.eks_version)
+
+    logger.info(
+        "EKS node AMI image version: %s",
+        parser.get_ami_version(args.versions_dir, args.eks_version),
+    )
+
+    images_wip_list = []
+
+    additional_images = parser.get_additional_images(args.versions_dir, args.eks_version)
+    docker_mappings = parser.get_docker_mappings(args.versions_dir, args.eks_version)
+    updated_additional_images = {}
+
+    for name, image in additional_images.items():
+        images_wip_list.append(image)
+        updated_additional_images[name] = f"{args.registry_prefix}{image}"
+
+    additional_images_json = {"additional_images": updated_additional_images}
+
+    workloads_data = parser.get_workloads(args.versions_dir, args.eks_version)
+
+    update_helm(args.update_helm, workloads_data)
+    # custom_chart_values = {}
+    parsed_charts = fetch_chart_info(workloads_data)
+    custom_chart_values = apply_chart_info(workloads_data, parsed_charts, args.registry_prefix, images_wip_list)
+
+    updated_images = []
+
+    for name, image in additional_images.items():
+        working_image = apply_image_mapping(image, docker_mappings)
+        updated_images.append({"src": working_image, "target": f"{args.registry_prefix}{image}"})
+    for image in images_wip_list:
+        working_image = apply_image_mapping(image, docker_mappings)
+        updated_images.append({"src": working_image, "target": f"{args.registry_prefix}{image}"})
 
     ami_json = {"ami": {"version": parser.get_ami_version(args.versions_dir, args.eks_version)}}
     charts_json = {"charts": custom_chart_values}
 
+    #  Add custom rules here....
+    try:
+        # cert-manager v1.6.x has removed the appVersion definition from crd...
```

Comment:
Does that mean we need to downgrade cert manager or update CRD? Manually manipulating the chart seems a bit unsafe